### ISSUE TITLE
[docs] incompatible with vuepress

### ISSUE THREAD

**Issue by Jinjiang** [vuejs/vuex#1275]
### Version
3.0.1

### Reproduction link
[https://github.com/Jinjiang/vuex/tree/vuepress](https://github.com/Jinjiang/vuex/tree/vuepress)

### Steps to reproduce
npm run docs

### What is expected?
Just try vuepress to dev/build docs

preview: https://5b01b3bb3672df726f83886e--objective-stonebraker-ca8353.netlify.com

### What is actually happening?
There always be some errors, until I changed some dev deps:

1. remove `rollup-plugin-buble`
2. remove (or upgrade v3 -> v4) `webpack`

by https://github.com/Jinjiang/vuex/commit/b74690d08cb43bff4eaa2031553e004922d802e2

But if I changed them, it failed with other npm scripts.

---
vuepress error branch:
[https://github.com/Jinjiang/vuex/tree/test](https://github.com/Jinjiang/vuex/tree/test)

vuepress error message:

```bash
> vuepress dev docs

Extracting site metadata...TypeError: Cannot destructure property `createHash` of 'undefined' or 'null'.
    at Object.<anonymous> (/Users/xxxxxxxx/vuex/node_modules/mini-css-extract-plugin/dist/index.js:26:44)
    at Module._compile (module.js:641:30)
    at Object.Module._extensions..js (module.js:652:10)
    at Module.load (module.js:560:32)
    at tryModuleLoad (module.js:503:12)
    at Function.Module._load (module.js:495:3)
    at Module.require (module.js:585:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/xxxxxxxx/vuex/node_modules/mini-css-extract-plugin/dist/cjs.js:3:18)
    at Module._compile (module.js:641:30)
    at Object.Module._extensions..js (module.js:652:10)
    at Module.load (module.js:560:32)
    at tryModuleLoad (module.js:503:12)
    at Function.Module._load (module.js:495:3)
    at Module.require (module.js:585:17)
    at require (internal/module.js:11:18)
```

I guess this should be considered as a bug of vuepress or we can do something in Vuex repo to walk around.

Thanks.

**Comment by yyx990803** [comment]
npm has a known issue when webpack 3 and webpack 4 coexist in the dep tree. We need to switch to Yarn.

**Comment by Jinjiang** [comment]
Still error after using yarn:

``` bash
$ npm run docs:build

> vuex@3.0.1 docs:build /Users/xxxxxxxx/vuex
> vuepress build docs


Compiling
(undefined) ./node_modules/vuepress/lib/app/.temp/routes.js
Module build failed: BubleLoaderError: 
6  :     {
7  :       path: "/",
8  :       component: Theme,
9  :       beforeEnter: (to, from, next) => {
10 :         import("/Users/xxxxxxxx/vuex/docs/README.md").then(comp => {
             ^
'import' and 'export' may only appear at the top level (10:8)
    at Object.module.exports (/Users/xxxxxxxx/vuex/node_modules/buble-loader/index.js:24:69)
 @ ./node_modules/vuepress/lib/app/app.js 7:0-37 45:0-6 55:12-18
 @ ./node_modules/vuepress/lib/app/clientEntry.js
 @ multi ./node_modules/vuepress/lib/app/clientEntry.js
(undefined) ./node_modules/vuepress/lib/app/.temp/routes.js
Module build failed: BubleLoaderError: 
6  :     {
7  :       path: "/",
8  :       component: Theme,
9  :       beforeEnter: (to, from, next) => {
10 :         import("/Users/xxxxxxxx/vuex/docs/README.md").then(comp => {
             ^
'import' and 'export' may only appear at the top level (10:8)
    at Object.module.exports (/Users/xxxxxxxx/vuex/node_modules/buble-loader/index.js:24:69)
 @ ./node_modules/vuepress/lib/app/app.js 7:0-37 45:0-6 55:12-18
 @ ./node_modules/vuepress/lib/app/serverEntry.js
 @ multi ./node_modules/vuepress/lib/app/serverEntry.js
Error: Failed to compile with errors.
    at webpack (/Users/xxxxxxxx/vuex/node_modules/vuepress/lib/build.js:103:18)
    at finalCallback (/Users/xxxxxxxx/vuex/node_modules/vuepress/node_modules/webpack/lib/MultiCompiler.js:241:39)
    at runWithDependencies.err (/Users/xxxxxxxx/vuex/node_modules/vuepress/node_modules/webpack/lib/MultiCompiler.js:259:6)
    at done (/Users/xxxxxxxx/vuex/node_modules/neo-async/async.js:2920:13)
    at runCompilers (/Users/xxxxxxxx/vuex/node_modules/vuepress/node_modules/webpack/lib/MultiCompiler.js:171:48)
    at err (/Users/xxxxxxxx/vuex/node_modules/vuepress/node_modules/webpack/lib/MultiCompiler.js:178:7)
    at compiler.run (/Users/xxxxxxxx/vuex/node_modules/vuepress/node_modules/webpack/lib/MultiCompiler.js:254:7)
    at finalCallback (/Users/xxxxxxxx/vuex/node_modules/vuepress/node_modules/webpack/lib/Compiler.js:156:39)
    at hooks.done.callAsync.err (/Users/xxxxxxxx/vuex/node_modules/vuepress/node_modules/webpack/lib/Compiler.js:172:13)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/xxxxxxxx/vuex/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:33:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/xxxxxxxx/vuex/node_modules/tapable/lib/Hook.js:35:21)
    at onCompiled (/Users/xxxxxxxx/vuex/node_modules/vuepress/node_modules/webpack/lib/Compiler.js:170:21)
    at hooks.afterCompile.callAsync.err (/Users/xxxxxxxx/vuex/node_modules/vuepress/node_modules/webpack/lib/Compiler.js:487:14)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/xxxxxxxx/vuex/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/xxxxxxxx/vuex/node_modules/tapable/lib/Hook.js:35:21)
    at compilation.seal.err (/Users/xxxxxxxx/vuex/node_modules/vuepress/node_modules/webpack/lib/Compiler.js:484:30)
```

**Comment by Jinjiang** [comment]
Finally, I still used npm, but upgraded `rollup-plugin-buble`, `webpack` and `vue-loader`, and fixed `/examples/webpack.config.js` + `/test/e2e/*`. All npm scripts works now.
Will create a PR for this.
Thanks.

**Comment by Jinjiang** [comment]
Fixed in #1277
